builtins.go

```diff
@@ -664,6 +665,63 @@ func builtinMd5(i *interpreter, trace traceElement, x value) (value, error) {
 	return makeValueString(hex.EncodeToString(hash[:])), nil
 }
 
+func builtinBase64(i *interpreter, trace traceElement, input value) (value, error) {
+	var byteArr []byte
+
+	var sanityCheck = func(v int) (string, bool) {
+		if !(v < 256 && v >= 0) {
+			msg := fmt.Sprintf("base64 encountered invalid codepoint value in the array (must be 0 <= X <= 255), got %d", v)
+			return msg, false
+		}
+
+		return "", true
+	}
+
+	vStr, err := i.getString(input, trace)
+	if err == nil {
```

Comment:
I don't like the assumption that any error returned by `i.getString` means simply that input was not a string. This function returns a generic error which may be anything in principle. If another error condition is currently possible or will be possible in the future versions, we'll have a really confusing bug here.

One more elegant approach would be to use a type switch here. Example: https://github.com/google/go-jsonnet/blob/v0.14.0/builtins.go#L44.

Reply:
that looks good. I'll update